### PR TITLE
Validation and field updates

### LINKED ISSUE #406: Data Model Configuration Tweaks
This is a placeholder for remaining data model configuration tweaks. This will be completed once a few things settle out and is intended as a landing spot for requested changes that are minor and won't hold things up if not done immediately.  

 - on the Repository Item Content Type make the `field_digital_publisher` a required field
 - On media, make `media_of` [a required field](https://github.com/jhu-idc/idc-isle-dc/pull/208), pointing to a repository object or collection

### PR DESCRIPTION
**Pull request by birkland**
* Makes `media_of` a required field.  Ingests will no longer silently orphan media if the `media_of` field is
empty, or malformed (i.e. a typo, pointing to a non-existent id).
* Makes `field_digital_publisher` required, as per stakeholder feedback related to citations
  * Multiple test data tweaks to accommodate this requirement
* Increases the length of alt title to 5000 characters.

Resolves jhu-idc/idc-general#406

### CODE REVIEW

File: codebase/config/sync/field.field.media.audio.field_media_of.yml [hunk 1]
```diff
@@ -5,9 +5,16 @@ dependencies:
   config:
     - field.storage.media.field_media_of
     - media.type.audio
+    - node.type.islandora_object
```
Comment (bseeger) [inline codebase/config/sync/field.field.media.audio.field_media_of.yml:8]: Hmmm...we might want this to include collections (collection_object) as well, so collections can have an image associated with them.

File: codebase/config/sync/field.field.media.audio.field_media_of.yml [hunk 2]
```diff
@@ -16,11 +23,17 @@ entity_type: media
...
 settings:
   handler: 'default:node'
-  handler_settings: {  }
+  handler_settings:
+    target_bundles:
+      islandora_object: islandora_object
+    sort:
+      field: _none
+    auto_create: false
+    auto_create_bundle: collection_object
```
Comment (bseeger) [inline codebase/config/sync/field.field.media.audio.field_media_of.yml:38]: I wonder what this line does? Some of the yml files have this filled in and some don't. (for the media).

File: codebase/config/sync/field.storage.media.field_media_of.yml [hunk 3]
```diff
@@ -6,8 +6,12 @@ dependencies:
     module:
       - islandora_core_feature
   module:
+    - field_permissions
```
Comment (bseeger) [inline codebase/config/sync/field.storage.media.field_media_of.yml:9]: Do you know what this does?  must be part of the media_of being required change.

**Review by bseeger — COMMENTED** [review]
There might be a need to change the media_of fields to accept collection_object as well. Testing this out now.

File: codebase/config/sync/field.field.media.audio.field_media_of.yml (hunk 1 above)
Comment (bseeger) [reply codebase/config/sync/field.field.media.audio.field_media_of.yml:8]: (if that is what this is doing?)

File: codebase/config/sync/field.field.media.audio.field_media_of.yml (hunk 2 above)
Comment (birkland) [reply codebase/config/sync/field.field.media.audio.field_media_of.yml:38]: unfortunately, I don't know.  Google found this:  https://www.drupal.org/project/drupal/issues/2412569

I can go check to make sure the "auto create" checkbox isn't checked.  If it is, I suspect it happened by default somehow.

File: codebase/config/sync/field.storage.media.field_media_of.yml (hunk 3 above)
Comment (birkland) [reply codebase/config/sync/field.storage.media.field_media_of.yml:9]: Unfortunately, I do not.

File: codebase/config/sync/field.field.media.audio.field_media_of.yml (hunk 2 above)
Comment (birkland) [reply codebase/config/sync/field.field.media.audio.field_media_of.yml:38]: The audio field definition doesn't seem any different from any of the others, especially in any way that makes `collection` stand out.  Interestingly, when I simply saved the page without making any changes, it added after `sort`:
```
     sort:
       field: _none
+      direction: ASC
     auto_create: false
 ```
 So sort by no field, ascending?  ` ¯\_(ツ)_/¯`

File: codebase/config/sync/field.field.media.audio.field_media_of.yml (hunk 1 above)
Comment (birkland) [reply codebase/config/sync/field.field.media.audio.field_media_of.yml:8]: I just pushed a commit to add that

File: codebase/config/sync/field.field.media.audio.field_media_of.yml (hunk 2 above)
Comment (bseeger) [reply codebase/config/sync/field.field.media.audio.field_media_of.yml:38]: weird, but probably innocuous.

**Comment by bseeger** [comment]
Spinning this up now to give it a look.

**Review by bseeger — APPROVED** [review]
This looks good to me and works as advertised. 

Worth nothing -- this ticket makes it so one can only attach Images to Collections.  One cannot attach other Media Types to Collections anymore, after this goes in.    
(I think this is sufficient, btw, but just wanted to get it in writing somewhere).

**Pull request merged**